tensorflow_quantum/python/util.py

```diff
@@ -63,9 +70,16 @@ def random_symbol_circuit(qubits,
             if isinstance(op, cirq.IdentityGate):
                 circuit[:i] += op.on(*locs)
             else:
+                this_random_symbol = sympy.Symbol(np.random.choice(symbols))
+                used_symbols.add(this_random_symbol)
                 circuit[:i] += (
                     op**((np.random.random() if include_scalars else 1.0) *
-                         sympy.Symbol(np.random.choice(symbols)))).on(*locs)
+                         this_random_symbol)).on(*locs)
+
+    # Use the rest of the symbols
+    for s in all_symbols - used_symbols:
+        circuit += cirq.Circuit(cirq.H(qubits[0])**s)
+
```

Comment:
We can probably do this a bit cleaner with something like this:

```python3
random_symbols = list(symbols)
random.shuffle(random_symbols)
location = 0
```
Then in the else:
```python3
circuit[:i] += op ** ((np.random.random() if include_scalars else 1.0) * sympy.Symbol(random_symbols[location % len(random_symbols)]))
location += 1
```

The padding at the end is very similar to the original approach. This way you don't have to do set lookup or set difference at all.

Reply:
Changed to remove the sets